tests/Imposter.CodeGenerator.Tests/Issues/Issue7/AssemblyAttributeWithGlobalInterfaceTests.cs

```diff
@@ -1,66 +1,119 @@
         GeneratorTestHelper.CreateContext(
-            Source,
+            source,
             baseSourceFileName: BaseSourceFileName,
             snippetFileName: SnippetFileName,
             assemblyName: nameof(AssemblyAttributeWithGlobalInterfaceTests),
             languageVersion: LanguageVersion.CSharp9
         );
 
     [Fact]
-    public async Task GivenAssemblyAttributeWithGlobalInterface_WhenImposterIsGenerated_ShouldBeAccessible()
+    public async Task GivenTargetInGlobalNamespaceWithSameNamespace_WhenGeneratorRuns_ShouldGenerateImposterInGlobalNamespace()
     {
-        var diagnostics = await CompileSnippet(
-            /*lang=csharp*/
-            """
-            using Imposter.Abstractions;
-            using Imposters.ITest;
-
-            public static class Scenario
-            {
-                public static void Execute()
-                {
-                    var imposter = new ITestImposter();
-                    var instance = imposter.Instance();
-                }
-            }
-            """
+        var testContext = await TestContextTask(SourceSameNamespace).ConfigureAwait(false);
+        var result = testContext.RunGenerator();
+        var compilation = testContext.Compilation;
+
+        var imposterSource = result.GeneratedSources.Single(source =>
+            source.HintName == "IGenericImposter.g.cs"
         );
 
-        GeneratorTestHelper.AssertNoDiagnostics(diagnostics);
+        var namespaceLine = imposterSource
+            .SourceText.ToString()
+            .Split([Environment.NewLine], StringSplitOptions.RemoveEmptyEntries)
+            .FirstOrDefault(line =>
+                line.TrimStart().StartsWith("namespace ", StringComparison.Ordinal)
+            );
+
+        namespaceLine.ShouldBeNull();
     }
 
-    private static async Task<ImmutableArray<Microsoft.CodeAnalysis.Diagnostic>> CompileSnippet(
-        string snippet
-    )
+    [Fact]
+    public async Task GivenTargetInGlobalNamespaceWithDedicatedNamespace_WhenGeneratorRuns_ShouldGenerateImposterInGlobalNamespace()
     {
-        var context = await TestContextTask.ConfigureAwait(false);
-        return context.CompileSnippet(snippet);
+        var testContext = await TestContextTask(SourceDedicatedNamespace).ConfigureAwait(false);
+        var result = testContext.RunGenerator();
+        var compilation = testContext.Compilation;
+
+        var imposterSource = result.GeneratedSources.Single(source =>
+            source.HintName == "IGenericImposter.g.cs"
+        );
+
+        var namespaceLine = imposterSource
+            .SourceText.ToString()
+            .Split([Environment.NewLine], StringSplitOptions.RemoveEmptyEntries)
+            .First(line => line.TrimStart().StartsWith("namespace ", StringComparison.Ordinal))
+            .Trim();
+
+        var expectedNamespace = $"namespace Imposters.{BuildSanitizedSuffix(compilation)}";
```

Comment:
i think we can simply hard code expected namespace here as opposed to replicating sanitization logic. thoughts ?

Reply:
Mm, I made the tests using `ImposterGeneratorNamespaceTests.cs` as a starting point. Look into it more closely, I guess that test uses it because it's playing with a generic type, so there's actually some sanitization logic involved. I copy/pasted that here, but it isn't really necessary. I'll update the tests to just use simple types.